common/common.cpp

```diff
@@ -1748,6 +1748,12 @@ std::string common_token_to_piece(const struct llama_vocab * vocab, llama_token
     return piece;
 }
 
+bool can_be_detokenized(const struct llama_context * ctx, const std::vector<llama_token> & tokens) {
```

Comment:
I don't think we should add a new API `can_detokenize` in `llama.h/common.h`

The simple solution is to check if a given token is within vocab before calling `common_token_to_piece`

Something like this:

```cpp
const int n_vocab = llama_vocab_n_tokens(vocab);
if (token >= n_vocab) {
  throw std::runtime_error("invalid token");
}
```

Reply:
I think the runtime_error won't be caught, which will still leave the DOS.

I removed the new API from llama.h and common.h, and instead made a function local to server.cpp with this line:
```
const int n_vocab = llama_vocab_n_tokens(vocab);
```